tests/save.js

```diff
@@ -28,47 +29,48 @@ function assertUniqueError(assert, err, dups, message) {
             return true;
         }
 
-        assert.equal(suberr.name, 'ValidatorError',
-            'each sub-error should be of type ValidatorError');
-        assert.equal(suberr.kind, 'Duplicate value',
-            'each sub-error\'s kind should be "Duplicate value"');
+
+        t.equal(suberr.name, 'ValidatorError',
+            'each sub-error should be of name ValidatorError');
+        t.equal(suberr.kind, 'unique',
+            'each sub-error\'s kind should be "unique"');
 
         if (message !== undefined) {
-            assert.equal(suberr.message, message,
+            t.equal(suberr.message, message,
                 'each sub-error should carry over the custom message');
         }
 
         // with buffer values, only compare the inner "buffer" property
-        assert.equal(
+        t.equal(
             suberr.properties.value.toString(), dups[key].toString(),
             'the sub-error should contain the duplicated value'
         );
 
-        assert.equal(
+        t.equal(
             suberr.properties.path, key,
             'the sub-error should contain the duplicated value\'s path'
         );
 
-        assert.equal(
-            suberr.properties.type, 'Duplicate value',
-            'the sub-error\'s type should be "Duplicate value"'
+        t.equal(
+            suberr.properties.kind, 'unique',
+            'the sub-error\'s kind should be "unique"'
```

Comment:
This is causing tests failure when you remove the `type` attribute from the `properties` object, because `suberr`’s own attributes are those translated by the ValidatorError constructor, and `suberr.properties` is a reference to the object passed in argument to that constructor.

I don’t actually understand `suberr.properties` purpose (and it seems redundant with `suberr`’s own attributes). We are left with two choices: removing this assertion altogether and ignoring `suberr.properties`, or using `suberr.properties.type` instead of `suberr.properties.kind` in this assertion.

The second choice looks best to me.